Raven.Database/FileSystem/Controllers/FilesStreamsController.cs

```diff
@@ -93,6 +99,93 @@ private void StreamToClient(Stream stream, int pageSize, Etag etag, OrderedPartC
             }
         }
 
+        [HttpPost]
+        [RavenRoute("fs/{fileSystemName}/streams/Export")]
+        public async Task<HttpResponseMessage> Export()
+        {
+            var filesJson = await ReadJsonAsync().ConfigureAwait(false);
+            var fileNames = filesJson.Value<RavenJArray>("FileNames").Values<string>().ToArray();
+            
+
+
+            var pushStreamContent = new PushStreamContent((stream, content, transportContext) => StreamExportToClient( stream, fileNames))
+            {
+                Headers =
+                {
+                    
+                    ContentType = new MediaTypeHeaderValue("application/json")
+                    {
+                        CharSet = "utf-8",
+                        
+                    },
+                    ContentEncoding = { "gzip" }
+                }
+            };
+
+            return new HttpResponseMessage(HttpStatusCode.OK)
+            {
+                Content = pushStreamContent
+            };
+        }
+
+        private void StreamExportToClient(Stream gzip2, string[] fileNames)
+        {
+            using (var gzip = new BufferedStream(gzip2))
```

Comment:
Why are you using buffered output here, and not actually gzip? The code seems to suggest that you would gzip it, and certainly that is what the content type you set says.